port to tvOS

Michelasso opened the pull request:
I have ported the framework to tvOS. I didn't touch the cocoapods or other additional stuff. I tested it in app I ported to tvOS as well, just adding the framework in a workspace, and it looks fine. There are no warnings either (I had issues with the umbrella headers but I fixed them changing the headers from "project" to "public").

tapwork commented:
Travis is complaining about different Swift Versions.
Could you use Swift 4.0 in your example?
Or 4.1 , but this would mean that we need to update Swift in the old example too

tapwork commented:
But thanks so far.
I need to have more time to through the PR. Could take a bit

Michelasso commented:
Oh crap. I though it was using the same Swift version or the original project. I am using Xcode 10 beta, (obviously. I need it because of the nice new TVUIKit framework for tvOS 12)... I'll take a look, it should be just a matter of modifying the Swift version to 4.0, test, commit and push again. 

Then I gave a fast look at the differences, and it changed the Cocoapod version as well (I have version 15.2 installed). But I know little to nothing about it. It has been a miracle that I managed to add it to Cocoapod immediately without breaking anything! I hope it won't matter.

Michelasso commented:
All right, I give up. Sorry for the PR spamming, but I have no idea why Travis keeps reporting the error "Malformed version number string". I though because i added the Swift version to the project, but removing it didn't fix it. Then I realised that I didn't increment the framework version number, so I raised it to 1.3.0. Still complaining. 

In https://github.com/CocoaPods/CocoaPods/issues/7765 they say that it should be an issue with RubyGems v2.7.7, fixed in Cocoapod 1.5.3. Maybe downgrading RubyGems in Travis to 2.7.6 will make it working. Or upgrading to Cocoapod 1.5.3, I don't know.  Unless you have any other suggestion to fix something on my side. Everything (both iOS and tvOS) seems working fine on my environment, so at least the core code must be ok.

tapwork commented on Classes/TWWatchdogInspector.m:
```diff
@@ -176,12 +176,21 @@ + (void)resetCountValues {
+#if TARGET_OS_IOS
+    CGRect statusBarFrame = [UIApplication sharedApplication].statusBarFrame;
+#elif TARGET_OS_TV
+    CGRect statusBarFrame = CGRectMake(0, 0, 1920, 20);
```
How about using `[UIScreen mainScreen].bounds.size.width` instead of `1920`

tapwork commented on SwiftExample/Podfile.lock:
```diff
@@ -1,16 +1,16 @@
 PODS:
-  - WatchdogInspector (1.2.0)
+  - WatchdogInspector (1.3.0)
```
You don't need to bump the version. I will do that later

tapwork commented on SwiftExample/SwiftExample_tvOS/AppDelegate.swift:
```diff
@@ -0,0 +1,48 @@
+//
+//  AppDelegate.swift
+//  SwiftExample_tvOS
```
Is it possible to use the existing example for `tvOS`? Would a bit cleaner and less files in the project

tapwork commented on WatchdogInspector.podspec:
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name     = 'WatchdogInspector'
-  s.version  = '1.2.0'
+  s.version  = '1.3.0'
```
You don't need to bump the version. I will do that later

Michelasso commented:
I have implemented all changes as requested. I have also made the height 1/54 of [UIScreen mainScreen].bounds.height, although I am not sure it should matter, it seems to me the tvOS UI always used the 1920x1080 resolution and then it scales it to the display resolution. But sure, now is correctly auto adjusting.

About the two versions of SwiftExample I did the most I could do. The resources are different, so I kept the common swift files and moved the independent resources to two separate folders. They both compile and run fine (iPad and Apple TV 4K with 4K display)!

Travis is still complaining, though. Is there something I can do on my side? Any idea what that error about malformed version is?

Michelasso commented:
Damn. I disabled the tvOS swift files but I forgot to delete them. Now it's fixed.

tapwork commented:
Thanks. Travis is still complaining about a broken podfile or podspec. I triggered it again. Can you can try running `pod install` on your local machine?

Michelasso commented:
I am running "pod install" in the SwiftExample folder, I hope that's correct because I can't find any other pod file. This is what I get:

```
SwiftExample Michele$ 
  [Ripristinato: 15 ago 2018, 11:59:10]
Last login: Wed Aug 15 11:59:09 on ttys005
SilverMoon:SwiftExample Michele$ pod install
Analyzing dependencies
Fetching podspec for `WatchdogInspector` from `../WatchdogInspector.podspec`
Downloading dependencies
Using WatchdogInspector (1.2.0)
Generating Pods project
Integrating client project
Sending stats
Pod installation complete! There is 1 dependency from the Podfile and 1 total pod installed.

[!] Automatically assigning platform `ios` with version `10.2` on target `SwiftExample` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.

[!] Automatically assigning platform `ios` with version `9.2` on target `SwiftExampleTests` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.

[!] Automatically assigning platform `tvos` with version `10.2` on target `SwiftExample_tvOS` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.

```

Maybe there is an issue because I cloned the git instead of installing it via cocoapod? I don't know, as I said I never really used cocoapod before. In case what should I do?

Michelasso commented:
No, what am I saying... I forked this version to make the pull requests, I cloned the initial one to add it to a demo app and there I made my initial modification (then ported to the current version). I use cocoapod only on this one, always running pod install before a commit.

tapwork commented:
@Michelasso could you make a `bundle update` in the root folder of the project. and commit the `Gemfile.lock` that should fix the bundler issue on travis

Michelasso commented:
@tapwork I am not sure about how to do a bundle update. Could you please give me the right command?

tapwork commented:
@Michelasso it's just the command `bundle update` in the terminal (in the root of the project)

Michelasso commented:
@tapwork oh, I see.. I thought it was a git command! I have done it. Let's see what Travis says, now. 😊

Michelasso commented:
Finally!! Ok, it should be fine, now! I just noticed that in tvOS I created a CGRect for nothing, while I just needed to define a CGSize. I may change it, or feel free to change it yourself. As it is now it works, so there is no hurry.

tapwork commented:
merged it...I am about to release a new pod version

Michelasso commented:
Ok, that sounds great! I have seen the changes in the master already. I believe the README.md file would need to be updated as well including the tvOS platform. 😉

tapwork commented:
thank you very mich again!! great stuff

Michelasso commented:
No probs! I am glad to contribute! I also learned few things.

PS: I see the README.md file includes tvOS already. Never mind, it must automatically detect it. Take care!
